File: 232. Implement Queue using Stacks/C++/Solution.cpp
class MyQueue {
    stack<int> stackA;
    stack<int> stackB;
public:
    /** Initialize your data structure here. */
    MyQueue() {

    }

    /** Push element x to the back of queue. */
    void push(int x) {
        stackA.push(x);
    }

    /** Removes the element from in front of queue and returns that element. */
    int pop() {
        if (stackB.empty()) {
            while (!stackA.empty()) {
                int item = stackA.top();
                stackB.push(item);
                stackA.pop();
            }
        }
        int result = stackB.top();
        stackB.pop();
        return result;
    }

    /** Get the front element. */
    int peek() {
        if (stackB.empty()) {
            while (!stackA.empty()) {
                int item = stackA.top();
                stackB.push(item);
                stackA.pop();
            }
        }
        return stackB.top();
    }

    /** Returns whether the queue is empty. */
    bool empty() {
        return stackA.empty() && stackB.empty();
    }
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue obj = new MyQueue();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.peek();
 * bool param_4 = obj.empty();
 */

File: 232. Implement Queue using Stacks/Python/Solution.py
class MyQueue(object):

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.stack_a = []
        self.stack_b = []

    def push(self, x):
        """
        Push element x to the back of queue.
        :type x: int
        :rtype: void
        """
        self.stack_a.append(x)

    def pop(self):
        """
        Removes the element from in front of queue and returns that element.
        :rtype: int
        """
        if self.stack_b:
            return self.stack_b.pop()
        else:
            while self.stack_a:
                self.stack_b.append(self.stack_a.pop())
            return self.stack_b.pop()

    def peek(self):
        """
        Get the front element.
        :rtype: int
        """
        if self.stack_b:
            return self.stack_b[-1]
        else:
            while self.stack_a:
                self.stack_b.append(self.stack_a.pop())
            return self.stack_b[-1]

    def empty(self):
        """
        Returns whether the queue is empty.
        :rtype: bool
        """
        return not self.stack_a and not self.stack_b


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()
